ndn-svs/logic.cpp

```diff
@@ -230,7 +246,7 @@ Logic::updateSeqNo(const SeqNo& seq, const NodeID& nid)
   }
 
   if (seq > prev)
-    sendSyncInterest();
+    retxSyncInterest(true, 0);
```

Comment:
Why? We should send sync interest immediately when new data is produced (i.e. sequence number updated).

Reply:
I think these have the same effect, right? Calling retxSyncInterest with true would cause the sync interest to be sent immediately. 
The reason I changed it was the call to retxSyncInterest(true, 0) would also reset the timer to m_retxDist. Supposed there is 5 seconds on the timer, if we just call sendSyncInterest(), we send a sync interest but leave the timer alone, so 5 seconds later, the timer goes off but the state vector hasn't changed since the last send. (I hope I'm understanding this correctly.
Also, it saved me from having to put "m_recording = false;" in multiple places since we should always exit out of the suppression state when we send.
Let me know what you think - I can change it back if you think that would be best.

Reply:
Fair. But then shouldn't `m_recording = false` be in `sendSyncInterest`? I notice you turn off the flag in `retxSyncInterest` even if the interest is not sent (it's outside the if condition).

Reply:
I think there is a condition where we would want to set m_recording = false without sending:
If we're in suppression state, (m_recording == true), and m_vv has nothing newer than m_recordedVv, (i.e: an interest came in that had missing state so we started recording but then a later interest came in that had the missing state) we would not want to send but we would still want to exit suppression state (set m_recording = false);

This logic does feel a bit tricky so if you can think of a way to simplify, let me know.

Reply:
Ah, yes.